test_conformance/integer_ops/test_integers.cpp

```diff
@@ -85,10 +85,23 @@ int test_single_param_integer_kernel(cl_command_queue queue, cl_context context,
                 sizeName );
 
     bool isOpenCL20Function = (strcmp(fnName,"ctz") == 0)? true: false;
+    size_t major = 0;
+    size_t minor = 0;
+    bool deviceLt20 = false;
+    error = get_device_version(deviceID, &major, &minor);
+    if (error != CL_SUCCESS)
+    {
+        vlog_error("\t\tFAILED -- Failed to get devcie version. (%d)\n", error);
+        return error;
+    }
+
+    if (major < 2) {
+        deviceLt20 = true;
+    }
```

Comment:
This approach won't work. The `integer_ctz` test (the only one to require 2.x AFAICT) will run and its kernel fail to build. The only change needed I think is to mark this test as requiring 2.x.

Also, for future reference, please use `get_device_cl_version` instead of `get_device_version` as this removes the need to handle errors.